cores/esp8266/Esp.h

```diff
@@ -23,6 +23,48 @@
 
 #include <Arduino.h>
 
+#ifndef PUYA_SUPPORT
+  #define PUYA_SUPPORT 1
```

Comment:
This means that the default is enabled, unless explicitly disabled. I suggest other way around: disabled, unless explicitly enabled. I prefer to avoid having this enabled by default by mistake for every board out there, so please change to 0.
I could be wrong, but the only boards today that seem to have the puya chip are ESP-01s and some ESP-7s, which look to be mislabeled ESP-01s or ESP-01 variants. With the above change, users who need the support can provide the define externally to the build, i.e.: -DPUYA_SUPPORT=1. A control could be added to the IDE menu, or users can simply add the flag locally in boards.txt.